### PR TITLE
Do not specify `-pthread` on macOS

### PR DESCRIPTION
**Pull request by SoapGentoo**
@nirbheek @ePirat checked with Clang/GCC and ICC on macOS.

### CODE REVIEW

File: run_unittests.py [hunk 1]
```diff
@@ -3270,17 +3270,16 @@ def test_pkgconfig_gen_deps(self):
...
 
         out = self._run(cmd + ['--cflags-only-other']).strip().split()
-        self.assertEqual(sorted(out), sorted(['-pthread', '-DCUSTOM']))
+        self.assertEqual(sorted(out), sorted(['-DCUSTOM'] + [] if is_osx() else ['-pthread']))
 
         out = self._run(cmd + ['--libs-only-l', '--libs-only-other']).strip().split()
-        self.assertEqual(sorted(out), sorted(['-pthread', '-lcustom',
-                                              '-llibmain', '-llibexposed']))
+        self.assertEqual(sorted(out), sorted(['-lcustom', '-llibmain', '-llibexposed']
+                                             + [] if is_osx() else ['-pthread']))
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__

[W503] line break before binary operator

File: run_unittests.py [hunk 2]
```diff
@@ -3270,17 +3270,16 @@ def test_pkgconfig_gen_deps(self):
...
 
         out = self._run(cmd + ['--libs-only-l', '--libs-only-other', '--static']).strip().split()
-        self.assertEqual(sorted(out), sorted(['-pthread', '-lcustom',
-                                              '-llibmain', '-llibexposed',
-                                              '-llibinternal', '-lcustom2',
-                                              '-lfoo']))
+        self.assertEqual(sorted(out), sorted(['-lcustom', '-llibmain', '-llibexposed',
+                                              '-llibinternal', '-lcustom2', '-lfoo']
+                                             + [] if is_osx() else ['-pthread']))
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__

[W503] line break before binary operator

**Comment by SoapGentoo** [comment]
Didn't see that #4303 fixed it already for all compilers.

**Comment by nirbheek** [comment]
This is a better fix than #4303 though.

**Pull request closed without merging**